src/Symfony/Component/Security/Core/Authorization/Voter/Vote.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization\Voter;
+
+/**
+ * A Vote is returned by a Voter and contains the access (granted, abstain or denied).
+ * It can also contain a reason explaining the vote decision.
+ *
+ * @author Dany Maillard <danymaillard93b@gmail.com>
+ */
+final class Vote
+{
+    /** @var int One of the VoterInterface::ACCESS_* constants */
+    private $access;
+    private $reason;
```

Comment:
maybe we should call it `$message` even if it is the `the ability for voter to return decision reason` ?

Reply:
sure, I don't have any problem with that. There will be a few places I have to change as well if you think this change is necessary. Personally I think reason makes sense, but I can see why you want to use message.